### PR TITLE
KOGITO-294 / KOGITO-296: [DMN Designer] Kogito - DMNMarshaller - Improve JsArrayLike-based APIs

### PR DESCRIPTION
**Pull request by karreiro**
See:
- https://issues.jboss.org/browse/KOGITO-294
- https://issues.jboss.org/browse/KOGITO-296

---

Part of an ensemble:
- https://github.com/gitgabrio/gwt-jsonix-marshallers/pull/11
- https://github.com/kiegroup/kie-wb-common/pull/2892

### CODE REVIEW

**Comment by karreiro** [comment]
Thanks for the review, @manstis :-)

We have two reasons for keeping the `getNativeXYZ()` for now.

1) Currently, our `List` APIs rely on those methods to work:
```java
@JsOverlay
public final List<JSITImport> getImport() {
    return JsUtils.toList(JsUtils.getUnwrappedElementsArray(getNativeImport()));
}
```

2) The getters provide read-only lists, so we still need to use native APIs for setting values in the lists:
```java
JsUtils.add(result.getNativeKnowledgeRequirement(), iReq);
```

Right now, I'm investigating if we can improve the API of the constructors. But, I believe that improving our setters it's a really important task as well (probably my next one, if you agree 😊). With our new setters working, we can change the `getNativeXYZ()` to `private`.

File: gwt-jsonix-schema-compiler/src/main/java/gwt/jsonix/marshallers/xjc/plugin/builders/ModelBuilder.java [hunk 1]
```diff
@@ -391,10 +382,60 @@ protected static String getPropertyClassName(CPropertyInfo toLog) {
         return toReturn;
     }
 
-    protected static void addGetter(JCodeModel jCodeModel, JDefinedClass jDefinedClass, JClass propertyRef, String
-            publicPropertyName, String privatePropertyName, String namespace) {
-        final JAnnotationUse name = BuilderUtils.addGetter(jCodeModel, jDefinedClass, propertyRef, publicPropertyName, privatePropertyName);
-        conditionalAddNamespaceToProperty(name, propertyRef, namespace);
+    private static void generateNativeGetter(final JCodeModel jCodeModel,
```
Comment (gitgabrio) [inline gwt-jsonix-schema-compiler/src/main/java/gwt/jsonix/marshallers/xjc/plugin/builders/ModelBuilder.java]: @karreiro 
Could you please rename this back to `addGetter` - to keep consistency with the `addSetter` ? (It does not matter if after that only the native or also the default getters will be generated) )

Comment (karreiro) [reply gwt-jsonix-schema-compiler/src/main/java/gwt/jsonix/marshallers/xjc/plugin/builders/ModelBuilder.java]: Fixed! Thanks :-)

**Comment by karreiro** [comment]
Thanks for the review, @gitgabrio. Changes applied.

I've added the new API we discussed this morning as well:

```java
@JsProperty(name = "_import", namespace = "TDefinitions")
public final native JsArrayLike<JSITImport> getNativeImport();

@JsOverlay
public final List<JSITImport> getImport() {
    return JsUtils.toList(JsUtils.getUnwrappedElementsArray(getNativeImport()));
}

@JsOverlay
public final void addImport(JSITImport element) {
    JsUtils.add(getNativeImport(), element);
}

@JsOverlay
public final void addAllImport(JSITImport... elements) {
    JsUtils.addAll(getNativeImport(), elements);
}

@JsOverlay
public final void removeImport(int index) {
    JsUtils.remove(getNativeImport(), index);
}
```

**Comment by karreiro** [comment]
@gitgabrio I've updated this branch considering the new changes I've pushed on master. See the new generated code:

```java
@JsProperty(name = "_import", namespace = "TDefinitions")
public final native JsArrayLike<JSITImport> getNativeImport();


@JsOverlay
public final List<JSITImport> getImport() {
    return JsUtils.toList(JsUtils.getUnwrappedElementsArray(getNativeImport()));
}

@JsOverlay
public final<D extends JSITImport >void addImport(final D element) {
    if (getNativeImport() == null) {
        setImport(JsUtils.getNativeArray());
    }
    JsUtils.add(getNativeImport(), element);
}

@JsOverlay
public final<D extends JSITImport >void addAllImport(D... elements) {
    if (getNativeImport() == null) {
        setImport(JsUtils.getNativeArray());
    }
    JsUtils.addAll(getNativeImport(), elements);
}


@JsOverlay
public final void removeImport(final int index) {
    JsUtils.remove(getNativeImport(), index);
}

@JsProperty(name = "_import", namespace = "TDefinitions")
public final native void setImport(JsArrayLike<JSITImport> _importParam);
```

**Comment by gitgabrio** [comment]
@manstis @karreiro 
D-O-N-E!!!

**Pull request merged**